docs/Developer_guide.md

```diff
@@ -33,19 +33,64 @@ The diagram below provides a summary representation of GovReady-Q's Django sitea
 
 ![Siteapp data model (not all tables represented)](assets/govready-q-siteapp-erd.png)
 
+GovReady-Q is multi-tenant. The siteapp data model represents users who are uniquely associated with an oragnization and have membership in different organization projects. Users must be invited to projects.
+
+Access control is based on organization and projects. Information cannot be shared acrossed organizations and only limited information can be shared across projects within an organization.
+
+
 ## Guidedmodules
 
 The diagram below provides a summary representation of GovReady-Q's Django guidedmodules data model that handles Compliance Apps, modules and questions, (tasks,) answers, and instrumentation. 
 
 ![Guildedmodules data model (not all tables represented)](assets/govready-q-guidedmodules-erd.png)
 
+The best way to understand the guidedmodule data model is to think of a questionnaire. A questionnaire contains multiple questions. Questions can be grouped into modules. A questionnaire and its modules can contain different types of questions. Each question has an answer.
+
+This description suggests a simple "questionnaire-question-answer" database model, yet the diagram does not seem to have tables for "Questionnaire" or "Question" or "Answer."
+
+That's because we have a few special demands for our questionnaire that will require a bit of creative thinking and a few abstractions. Some of the additional requirements are:
+
+* blank questionaires are reusable and can be easily loaded into different installs of the database at different organizations;
+* anyone can author questionnaires and blank questionnaires can be kept private or shared publicly;
+* one questionnaire's answers can be accessed and used by another questionnaire if the questionnaires are part of the same project;
+* arbitrary questionnaires can be associated with the same project; so we won't know up front which answers can be shared;
+* support a question type whose answer is another questionnaire;
+* allow blank questionnaires to be versioned, answered questionnaires to be updated in non-destructive ways, and to preserve answered questionnaires over the course of years,
+* allow the answer to questions to change while preserving a complete history of answers,
+* support assigning questionnaires and questions to different users to answer.
+
+The first bit of creative thinking is to collapse the concepts of "questionnaire" and "module." A questionnaire is a module that contains a list of questions or a list of modules. We represent both "questionnaires" and "modules" as "modules."
```

Comment:
I didn't really follow this paragraph and am not sure it's right. I think a Questionnaire is an AppInstance.

Reply:
I will do that and submit a new version with the questionnaires equated to apps.